lib/pub_sub/subscriber.rb

```diff
@@ -66,5 +66,19 @@ def queue_arn(region)
     def queue_url(region)
       sqs(region).create_queue(queue_name: PubSub.service_identifier).queue_url
     end
+
+    # Collects all :topics or :subscriptions for a given SNS client
+    def self.collect_all(client, collection_type)
```

Comment:
This feels a bit out of place as a class method. Seems like a helper method that isn't necessarily related to the rest of this class. What do you think about adding this a instance methods. 

`#topics(*regions)` returning either an array of topics or a hash of region to topic mapping `{ region_1: [ *topics ], region_2: [ *topics ] }`